exchanges/account/account.go

```diff
@@ -268,22 +268,34 @@ func (s *Service) Update(incoming *Holdings, creds *Credentials) error {
 
 // load checks to see if there is a change from incoming balance, if there is a
 // change it will change then alert external routines.
-func (b *ProtectedBalance) load(change Balance) {
+func (b *ProtectedBalance) load(change *Balance) bool {
 	b.m.Lock()
 	defer b.m.Unlock()
+	if change == nil {
+		return false
+	}
+	if b.updatedAt.After(change.UpdatedAt) {
+		return false
+	}
+	if !b.updatedAt.IsZero() && b.updatedAt == change.UpdatedAt {
```

Comment:
Can use equal method.
```suggestion
	if !b.updatedAt.IsZero() && b.updatedAt.Equal(change.UpdatedAt) {
```

Also suggestion we can probably merge the above if case with `b.updatedAt.After(change.UpdatedAt)`

```go
if !b.updatedAt.IsZero() {
		if b.updatedAt.After(change.UpdatedAt) {
			return false
		}
		if b.updatedAt.Equal(change.UpdatedAt) {
		return false
	}
}
```

Reply:
I merged the above.
```
	if !b.updatedAt.IsZero() && !b.updatedAt.Before(change.UpdatedAt) {
		return errOutOfSequence
	}
```